packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -242,7 +237,11 @@ export interface SdkDictionaryType extends SdkTypeBase {
   kind: "dict";
   keyType: SdkType;
   valueType: SdkType;
-  nullableValues: boolean;
+}
+
+export interface SdkNullableType extends SdkTypeBase {
+  kind: "nullable";
+  valueType: SdkType;
```

Comment:
could we have another name for this?
Because in this context, this is never a value, not like in the SdkArrayType or SdkDictionaryType, that thing is called "the value of this array/dictionary" therefore it is `valueType`, but here it makes no sense that we say "we have a value inside a null type".

How about just `type` or `innerType`?

Similar comment also applies to `SdkUnionType`, we have `values` but those are not the "values". `valueTypes` makes more sense but not so much, similar with above reason - in an union of `string | int`, the `string` or `int` are not called as `value` of the union, in this case, `variantTypes` might be a better name.

Reply:
i'm ok to switch it back to `type`, and I get what you're saying about `union` as well. Can you open a separate issue for what you'd like to see in unions, and we can do the correct deprecation?